src/main/java/no/unit/nva/cristin/projects/CristinApiClient.java

```diff
@@ -138,14 +141,64 @@ protected CristinProject getProject(String id, String language) throws ApiGatewa
         throws ApiGatewayException {
 
         List<CristinProject> projectsFromQuery = asList(getDeserializedResponse(response, CristinProject[].class));
+        List<URI> cristinUris = extractCristinUrisFromProjects(language, projectsFromQuery);
+        List<HttpResponse<String>> individualResponses = fetchQueryResultsOneByOne(cristinUris);
+        List<CristinProject> enrichedCristinProjects = mapValidResponsesToCristinProjects(individualResponses);
+        enrichedCristinProjects =
+            combineResultsWithQueryInCaseEnrichmentFails(projectsFromQuery, enrichedCristinProjects);
 
+        return enrichedCristinProjects;
+    }
+
+    protected List<CristinProject> combineResultsWithQueryInCaseEnrichmentFails(
+        List<CristinProject> projectsFromQuery,
+        List<CristinProject> enrichedProjects) {
+
+        if (projectsFromQuery.size() == enrichedProjects.size()) {
+            return enrichedProjects;
+        }
+
+        List<CristinProject> missingProjects = projectsFromQuery.stream()
+            .filter(queryProject -> enrichedProjects.stream().noneMatch(
+                enrichedProject -> enrichedProject.getCristinProjectId().equals(queryProject.getCristinProjectId())))
+            .collect(Collectors.toList());
+
+        enrichedProjects.addAll(missingProjects);
+
+        return enrichedProjects;
+    }
+
+    // TODO: Split and test
+    @JacocoGenerated
+    protected List<HttpResponse<String>> fetchQueryResultsOneByOne(List<URI> uris) {
+        List<CompletableFuture<HttpResponse<String>>> responsesContainer =
+            uris.stream().map(uri ->
+                client.sendAsync(
+                    HttpRequest.newBuilder(uri).GET().build(),
+                    BodyHandlers.ofString(StandardCharsets.UTF_8)))
+                .collect(Collectors.toList());
+
+        return responsesContainer.stream()
+            .map(attempt(CompletableFuture::get))
+            .map(Try::get)
+            .filter(this::isSuccessfulRequest)
+            .collect(Collectors.toList());
+    }
+
+    protected boolean isSuccessfulRequest(HttpResponse<String> response) {
+        try {
+            checkHttpStatusCode(response.uri().toString(), response.statusCode());
+            return true;
+        } catch (Exception ex) {
+            return false;
+        }
+    }
+
+    private List<URI> extractCristinUrisFromProjects(String language, List<CristinProject> projectsFromQuery) {
         return projectsFromQuery.stream()
-            .map(project -> attempt(() -> getProject(project.getCristinProjectId(), language))
```

Comment:
old enrichment `getProject` which could not be used as `CompletableFuture` has to do everything in one operation. Also compare line number to see where it was before PR